### ISSUE TITLE
locals() should signal usage

### ISSUE THREAD

**Issue by memeplex** [microsoft/pyright#1955]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**

Using pylance I get warnings about not accessed parameters even if I capture the parameters using locals() for further usage. 

An alternative pattern would be to put kwargs in the function signature, but that way a lot of the "contract" is lost. Sometimes you just need to work on the arguments as a bundle, for example in order to forward them to another function or to override some of them using another dictionary, etc. But this is an implementation detail that shouldn't be exposed as a dumbed down signature. And having a lot of false positives turns the static checker less useful.

**To Reproduce**

```
def f(x, y, z):
    kwargs = locals()
    ...
```

This obviously applies to other previously defined local variables, not only arguments.

**Expected behavior**

Calling `locals()` signals usage of the previously defined local environment, so no warning should be shown. 

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool? Which version? You can find the version of the VS Code extension by clicking on the Pyright icon in the extensions panel.

Pylance 2021.6.0
VSCode 1.56.2

**Additional context**

**Comment by erictraut** [comment]
Pyright tracks explicit references to symbols. There are many examples of implicit references. You've highlighted one here. We've decided not to try to cover implicit references, since the list of special cases is potentially unbounded.

You mentioned that you get warnings about not accessed parameters. I'm a bit confused by this, since we never emit an error or warning for unreferenced parameters. We do emit a hint to the editor so it can display the symbol using a subtle "grayed out" appearance, but you should never see an error or warning here. If you are seeing an actual warning, let us know.

**Comment by memeplex** [comment]
Yes, you are right, I didn't mean "warnings" in the precise technical way you mention, it's just different highlighting in the editor, not a warning message in the list of problems. I understand the decision not to cover implicit references at all, you have a point there. Alternatively, is there a way to turn off that hinting for an entire signature or function? I wasn't able to figure that out from the documentation or by trial&error.

**Comment by erictraut** [comment]
No, there's no way to disable these hints. That's intentional, since it's not a diagnostic but meant to be just a subtle visual clue to the programmer that there is not explicit reference.

**Comment by memeplex** [comment]
Ok, great then, thanks!
